In [117]:
import pandas as pd 
import numpy as np

df = pd.read_csv("data/data.csv", sep=';', decimal=",", encoding='cp1252')
pd.set_option('display.max_columns',None)
df.columns = df.columns.str.strip()
df.head()

,Group,Sex,Age,Patients number per hour,Arrival mode,Injury,Chief_complain,Mental,Pain,NRS_pain,SBP,DBP,HR,RR,BT,Saturation,KTAS_RN,Diagnosis in ED,Disposition,KTAS_expert,Error_group,Length of stay_min,KTAS duration_min,mistriage
0,2,2,71,3,3,2,right ocular pain,1,1,2,160,100,84,18,36.6,100,2,Corneal abrasion,1,4,2,86,5.00,1
1,1,1,56,12,3,2,right forearm burn,1,1,2,137,75,60,20,36.5,NaN,4,"Burn of hand, firts degree dorsum",1,5,4,64,3.95,1
2,2,1,68,8,2,2,"arm pain, Lt",1,1,2,130,80,102,20,36.6,98,4,"Fracture of surgical neck of humerus, closed",2,5,4,862,1.00,1
3,1,2,71,8,1,1,ascites tapping,1,1,3,139,94,88,20,36.5,NaN,4,Alcoholic liver cirrhosis with ascites,1,5,6,108,9.83,1
4,1,2,58,4,3,1,"distension, abd",1,1,3,91,67,93,18,36.5,NaN,4,Ascites,1,5,8,109,6.60,1


In [118]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1267 entries, 0 to 1266
Data columns (total 24 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Group                     1267 non-null   int64  
 1   Sex                       1267 non-null   int64  
 2   Age                       1267 non-null   int64  
 3   Patients number per hour  1267 non-null   int64  
 4   Arrival mode              1267 non-null   int64  
 5   Injury                    1267 non-null   int64  
 6   Chief_complain            1267 non-null   str    
 7   Mental                    1267 non-null   int64  
 8   Pain                      1267 non-null   int64  
 9   NRS_pain                  1267 non-null   str    
 10  SBP                       1267 non-null   str    
 11  DBP                       1267 non-null   str    
 12  HR                        1267 non-null   str    
 13  RR                        1267 non-null   str    
 14  BT                 

In [119]:
df.tail()

,Group,Sex,Age,Patients number per hour,Arrival mode,Injury,Chief_complain,Mental,Pain,NRS_pain,SBP,DBP,HR,RR,BT,Saturation,KTAS_RN,Diagnosis in ED,Disposition,KTAS_expert,Error_group,Length of stay_min,KTAS duration_min,mistriage
1262,2,2,79,5,2,1,mental change,1,0,#BOÞ!,120,80,86,22,36.4,95,2,Cerebral infarction due to unspecified occlusi...,2,2,0,1995,3.0,0
1263,2,2,81,2,3,1,dysuria,1,0,#BOÞ!,120,80,94,20,36.4,97,4,Dysuria,1,4,0,1000,2.0,0
1264,2,2,81,17,2,1,dizziness,1,0,#BOÞ!,130,90,80,20,36.2,99,3,Dizziness and giddiness,1,3,0,310,4.0,0
1265,2,1,81,2,2,2,"Sensory, Decreased",1,0,#BOÞ!,170,100,78,20,36.6,98,3,"Cord compression, unspecified",7,3,0,475,5.0,0
1266,2,2,87,8,4,1,orthopnea,1,0,#BOÞ!,150,80,62,20,36.2,98,3,Dyspnoea,2,3,0,887,2.0,0


Task 1. 
What kind of information does this dataset contain?
    It contains a patient labelled from 0 to 1266 as well as some variable collected during their ED visit. This includes vital signs, their KTAS, sex, age, injury, pain, and chief_complaint
What does one row represent?
    A patient and their variables
What does one column represent?
    One type of variable
    

In [120]:
df.shape
print("Number of rows:",df.shape[0])
print("Numbers of columns:",df.shape[1])
df.columns.tolist()

Number of rows: 1267
Numbers of columns: 24


['Group',
 'Sex',
 'Age',
 'Patients number per hour',
 'Arrival mode',
 'Injury',
 'Chief_complain',
 'Mental',
 'Pain',
 'NRS_pain',
 'SBP',
 'DBP',
 'HR',
 'RR',
 'BT',
 'Saturation',
 'KTAS_RN',
 'Diagnosis in ED',
 'Disposition',
 'KTAS_expert',
 'Error_group',
 'Length of stay_min',
 'KTAS duration_min',
 'mistriage']

This dataset contains a total of 1267 rows and 24 columns.
Each row represents one emergency department patient record.
Each column represents a type of variable taken from the patient's record

In [121]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1267 entries, 0 to 1266
Data columns (total 24 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Group                     1267 non-null   int64  
 1   Sex                       1267 non-null   int64  
 2   Age                       1267 non-null   int64  
 3   Patients number per hour  1267 non-null   int64  
 4   Arrival mode              1267 non-null   int64  
 5   Injury                    1267 non-null   int64  
 6   Chief_complain            1267 non-null   str    
 7   Mental                    1267 non-null   int64  
 8   Pain                      1267 non-null   int64  
 9   NRS_pain                  1267 non-null   str    
 10  SBP                       1267 non-null   str    
 11  DBP                       1267 non-null   str    
 12  HR                        1267 non-null   str    
 13  RR                        1267 non-null   str    
 14  BT                 

numeric columns: age, KTAS duration_min, length of stay_min
string/catergorical columns:chief complain, Diagnosis in ED
columns that contain numbers but actually represent categories: sex,group,arrival mode,mental,pain,disposition,KTAS

In [122]:
df.describe()
print(f'Mean age: {df["Age"].mean():.2f}')
print(f'Median age:" {df["Age"].median():.2f}')
print(f'Minimum age:" {df["Age"].min():.2f}')
print(f'Maximum age:" {df["Age"].max():.2f}')

Mean age: 54.42
Median age:" 57.00
Minimum age:" 16.00
Maximum age:" 96.00


In [123]:
df["Chief_complain"].head(10)

0        right ocular pain
1       right forearm burn
2             arm pain, Lt
3          ascites tapping
4          distension, abd
5                    fever
6    With chest discomfort
7              pain, chest
8      LBP - Low back pain
9           Eczema, Eyelid
Name: Chief_complain, dtype: str

In [124]:
df["Age"].head(10)

0    71
1    56
2    68
3    71
4    58
5    54
6    49
7    78
8    32
9    38
Name: Age, dtype: int64

In [125]:
df[["Age", "Sex", "SBP", "HR"]].head(6)

,Age,Sex,SBP,HR
0,71,2,160,84
1,56,1,137,60
2,68,1,130,102
3,71,2,139,88
4,58,2,91,93
5,54,1,140,94


In [126]:
elderly = df[df["Age"] >= 65]
print("Number of patients aged 65 of older:",len(elderly))
elderly.head()

Number of patients aged 65 of older: 467


,Group,Sex,Age,Patients number per hour,Arrival mode,Injury,Chief_complain,Mental,Pain,NRS_pain,SBP,DBP,HR,RR,BT,Saturation,KTAS_RN,Diagnosis in ED,Disposition,KTAS_expert,Error_group,Length of stay_min,KTAS duration_min,mistriage
0,2,2,71,3,3,2,right ocular pain,1,1,2,160,100,84,18,36.6,100,2,Corneal abrasion,1,4,2,86,5.00,1
2,2,1,68,8,2,2,"arm pain, Lt",1,1,2,130,80,102,20,36.6,98,4,"Fracture of surgical neck of humerus, closed",2,5,4,862,1.00,1
3,1,2,71,8,1,1,ascites tapping,1,1,3,139,94,88,20,36.5,NaN,4,Alcoholic liver cirrhosis with ascites,1,5,6,108,9.83,1
7,1,2,78,14,3,1,"pain, chest",1,1,3,169,86,80,20,36,NaN,2,Acute coronary syndrome,1,3,2,247,10.23,1
15,2,1,65,12,4,1,headache,1,1,3,140,100,80,20,36.0,98,3,"Subarachnoid haemorrhage, unspecified",2,4,4,153,3.00,1


In [127]:
df["HR"] = pd.to_numeric(df["HR"], errors = "coerce")
high_hr = df[df["HR"] >= 100]
print("Number of patients with HR >= 100:", len(high_hr))
high_hr.head()

Number of patients with HR >= 100: 250


,Group,Sex,Age,Patients number per hour,Arrival mode,Injury,Chief_complain,Mental,Pain,NRS_pain,SBP,DBP,HR,RR,BT,Saturation,KTAS_RN,Diagnosis in ED,Disposition,KTAS_expert,Error_group,Length of stay_min,KTAS duration_min,mistriage
2,2,1,68,8,2,2,"arm pain, Lt",1,1,2,130,80,102.0,20,36.6,98,4,"Fracture of surgical neck of humerus, closed",2,5,4,862,1.00,1
12,2,1,53,10,3,1,epigastric pain,1,1,3,120,80,106.0,20,36.2,98,3,Infectious or septic colitis,1,4,4,8067,5.00,1
14,1,2,58,4,3,1,headache,1,1,3,135,82,106.0,20,37.6,NaN,3,Infective endocarditis,2,4,4,1318,9.88,1
17,2,2,53,12,2,2,Open Wound,1,1,4,130,80,100.0,20,36.4,97,2,"Open wound of other parts of head, face",2,3,2,613,4.00,1
19,1,2,38,3,2,1,Gingival swelling,1,1,4,133,93,109.0,16,36.5,NaN,3,Odontogenic infection,1,4,4,349,12.45,1


In [128]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1267 entries, 0 to 1266
Data columns (total 24 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Group                     1267 non-null   int64  
 1   Sex                       1267 non-null   int64  
 2   Age                       1267 non-null   int64  
 3   Patients number per hour  1267 non-null   int64  
 4   Arrival mode              1267 non-null   int64  
 5   Injury                    1267 non-null   int64  
 6   Chief_complain            1267 non-null   str    
 7   Mental                    1267 non-null   int64  
 8   Pain                      1267 non-null   int64  
 9   NRS_pain                  1267 non-null   str    
 10  SBP                       1267 non-null   str    
 11  DBP                       1267 non-null   str    
 12  HR                        1247 non-null   float64
 13  RR                        1267 non-null   str    
 14  BT                 

In [129]:
df["Saturation"] = pd.to_numeric(df["Saturation"], errors = "coerce")
low_spo2 = df[df["Saturation"] < 95]
print("Number of patients with oxygen saturation below 95:", len(low_spo2))
low_spo2.head(2)

Number of patients with oxygen saturation below 95: 47


,Group,Sex,Age,Patients number per hour,Arrival mode,Injury,Chief_complain,Mental,Pain,NRS_pain,SBP,DBP,HR,RR,BT,Saturation,KTAS_RN,Diagnosis in ED,Disposition,KTAS_expert,Error_group,Length of stay_min,KTAS duration_min,mistriage
61,2,1,70,8,2,1,dyspnea,2,1,3,??,??,96.0,22,36.3,78.0,2,Dyspnoea,6,1,1,593,2.0,2
132,2,2,82,6,4,1,dyspnea,1,0,#BOÞ!,150,100,116.0,26,37.0,68.0,2,"Bacterial pneumonia, unspecified",2,1,1,9270,2.0,2


In [130]:
df["Saturation"] = pd.to_numeric(df["Saturation"], errors = "coerce")
result = df[(df["Saturation"] < 95) & (df["Age"] >= 65)]
print("Number of senior patients with oxygen saturation below 95:", len(result))
result.head()

Number of senior patients with oxygen saturation below 95: 32


,Group,Sex,Age,Patients number per hour,Arrival mode,Injury,Chief_complain,Mental,Pain,NRS_pain,SBP,DBP,HR,RR,BT,Saturation,KTAS_RN,Diagnosis in ED,Disposition,KTAS_expert,Error_group,Length of stay_min,KTAS duration_min,mistriage
61,2,1,70,8,2,1,dyspnea,2,1,3,??,??,96.0,22,36.3,78.0,2,Dyspnoea,6,1,1,593,2.0,2
132,2,2,82,6,4,1,dyspnea,1,0,#BOÞ!,150,100,116.0,26,37.0,68.0,2,"Bacterial pneumonia, unspecified",2,1,1,9270,2.0,2
134,2,1,92,9,3,1,dyspnea,1,0,#BOÞ!,150,90,110.0,26,36.5,88.0,3,Other specified chronic obstructive pulmonary ...,2,2,1,699091,4.0,2
140,2,2,73,8,3,1,dyspnea,1,0,#BOÞ!,140,80,86.0,24,36.4,89.0,3,NaN,1,2,2,518,2.0,2
153,2,2,65,7,2,2,mental change,3,0,#BOÞ!,70,??,100.0,20,36.0,94.0,2,Traumatic subdural hemorrhage without open int...,5,1,5,150,2.0,2


In [131]:
df.sort_values("Age",ascending=False).head(10)

,Group,Sex,Age,Patients number per hour,Arrival mode,Injury,Chief_complain,Mental,Pain,NRS_pain,SBP,DBP,HR,RR,BT,Saturation,KTAS_RN,Diagnosis in ED,Disposition,KTAS_expert,Error_group,Length of stay_min,KTAS duration_min,mistriage
1209,2,2,96,7,4,1,general weakness,1,0,#BOÞ!,90,60,68.0,20,36.2,91.0,2,Pericardial effusion (noninflammatory),2,2,0,20076,4.00,0
1208,1,1,95,9,7,1,Hyperkalemia (disorder),1,0,#BOÞ!,137,71,58.0,18,36.8,NaN,3,Ascites,1,3,0,0,16.00,0
448,2,1,94,5,3,1,"abdominal pain, LUQ",1,1,3,170,100,68.0,20,36.1,99.0,4,Unspecified abdominal pain,1,4,0,283,2.00,0
1207,1,2,93,12,1,1,dizziness,1,0,#BOÞ!,121,70,76.0,18,36.6,NaN,3,Dizziness,1,3,0,0,15.00,0
134,2,1,92,9,3,1,dyspnea,1,0,#BOÞ!,150,90,110.0,26,36.5,88.0,3,Other specified chronic obstructive pulmonary ...,2,2,1,699091,4.00,2
809,1,2,91,12,4,1,fever,3,0,#BOÞ!,142,86,110.0,24,38.6,NaN,2,Aspiration pneumonia,2,2,0,1,4.00,0
1206,1,2,91,11,6,1,medication,1,0,#BOÞ!,131,76,62.0,18,36.5,NaN,5,Benign prostatic hyperplasia with obstruction,1,5,0,80,6.30,0
133,1,1,90,9,2,1,acute dyspnea,1,1,#BOÞ!,65,45,100.0,30,36,NaN,2,"Colon cancer, sigmoid",5,1,1,126,9.58,2
49,2,2,89,8,4,1,melena,1,0,#BOÞ!,100,60,74.0,20,36.2,95.0,2,"Acute with haemorrhage, duodenal ulcer",7,3,7,8397,4.00,1
554,2,1,89,3,2,1,Left chest pain,1,1,4,100,60,48.0,20,36.0,95.0,2,Other chest pain,1,2,0,148,2.00,0


In [132]:
df.sort_values("HR",ascending=False).head(10)

,Group,Sex,Age,Patients number per hour,Arrival mode,Injury,Chief_complain,Mental,Pain,NRS_pain,SBP,DBP,HR,RR,BT,Saturation,KTAS_RN,Diagnosis in ED,Disposition,KTAS_expert,Error_group,Length of stay_min,KTAS duration_min,mistriage
454,1,1,85,10,3,1,palpitation,1,1,4,118,69,148.0,20,37.3,NaN,3,Atrial flutter,2,3,0,296,7.23,0
32,1,1,74,8,3,1,palpitation,1,1,#BOÞ!,91,74,145.0,18,36.8,NaN,2,Paroxysmal supraventricular tachycardia,1,3,1,95,8.10,1
152,2,2,63,9,4,1,mental change,3,0,#BOÞ!,110,70,140.0,24,36.2,92.0,3,"Encephalopathy, unspecified",5,2,5,8188,4.00,2
525,1,1,60,10,3,1,abd pain,1,1,4,198,117,137.0,18,36.6,NaN,3,Urinary retension,1,3,0,280,8.33,0
1108,1,2,72,14,2,1,rigth side weakness,1,0,#BOÞ!,204,98,137.0,18,38,NaN,3,Metastatic cancer to the brain,2,3,0,1437,11.02,0
370,1,1,59,5,3,1,LLQ pain,1,1,3,142,89,135.0,20,36.8,NaN,3,Constipation,1,3,0,76,7.00,0
1050,1,1,65,7,3,1,palpitation,1,0,#BOÞ!,143,87,131.0,20,36,NaN,3,Palpitation,1,3,0,110,9.53,0
129,1,2,67,9,3,1,fever & chill,1,0,#BOÞ!,143,70,130.0,20,38.1,NaN,4,Fever,1,2,1,267,3.60,2
621,1,2,54,5,3,1,abdomen pain,1,1,5,137,77,129.0,20,37.8,NaN,3,Metastatic cancer to the liver,2,3,0,478,17.23,0
620,1,2,54,5,3,1,abdomen pain,1,1,5,137,77,129.0,20,37.8,NaN,3,Liver abscess,2,3,0,478,17.23,0


In [133]:
df.sort_values("Saturation",ascending=True).head(10)

,Group,Sex,Age,Patients number per hour,Arrival mode,Injury,Chief_complain,Mental,Pain,NRS_pain,SBP,DBP,HR,RR,BT,Saturation,KTAS_RN,Diagnosis in ED,Disposition,KTAS_expert,Error_group,Length of stay_min,KTAS duration_min,mistriage
1148,2,2,77,7,2,1,dizziness,1,0,#BOÞ!,130,80,56.0,20,36.2,20.0,3,Horizontal canal benign positional vertigo,1,3,0,8258,1.0,0
132,2,2,82,6,4,1,dyspnea,1,0,#BOÞ!,150,100,116.0,26,37.0,68.0,2,"Bacterial pneumonia, unspecified",2,1,1,9270,2.0,2
1167,2,1,79,8,4,1,blood tinged sputum,1,0,#BOÞ!,130,80,60.0,24,36.4,74.0,2,Haemoptysis,6,2,0,9378,5.0,0
434,2,2,80,10,3,1,dyspnea,1,1,3,120,80,118.0,20,36.0,76.0,2,Acute and subacute infective endocarditis,2,2,0,9984,4.0,0
61,2,1,70,8,2,1,dyspnea,2,1,3,??,??,96.0,22,36.3,78.0,2,Dyspnoea,6,1,1,593,2.0,2
988,2,1,54,3,3,1,cough,1,0,#BOÞ!,180,100,108.0,24,36.6,80.0,3,Dyspnoea,2,3,0,10895,4.0,0
845,2,2,22,10,2,1,mental change,3,0,#BOÞ!,100,60,98.0,24,37.6,85.0,2,Stupor,1,2,0,8703,4.0,0
911,2,2,39,10,4,1,Confusion state,3,0,#BOÞ!,80,50,72.0,20,36.8,86.0,2,Hepatorenal syndrome,2,2,0,518,5.0,0
968,2,1,51,1,4,1,mental change,4,0,#BOÞ!,60,??,72.0,??,35.7,86.0,1,Cardiac arrest with successful resuscitation,6,1,0,392,5.0,0
134,2,1,92,9,3,1,dyspnea,1,0,#BOÞ!,150,90,110.0,26,36.5,88.0,3,Other specified chronic obstructive pulmonary ...,2,2,1,699091,4.0,2


In [134]:
df [["Age","Sex","HR","Saturation"]].sort_values("Age",ascending=False).head(10)

,Age,Sex,HR,Saturation
1209,96,2,68.0,91.0
1208,95,1,58.0,NaN
448,94,1,68.0,99.0
1207,93,2,76.0,NaN
134,92,1,110.0,88.0
809,91,2,110.0,NaN
1206,91,2,62.0,NaN
133,90,1,100.0,NaN
49,89,2,74.0,95.0
554,89,1,48.0,95.0


In [135]:
df["Sex"].value_counts()

Sex
2    661
1    606
Name: count, dtype: int64

In [ ]:
df["KTAS_expert"].value_counts()

KTAS_expert
3    487
4    459
2    220
5     75
1     26
Name: count, dtype: int64

1. What is the name of the downloaded CSV file?
    data.csv
2. How many rows and columns are in the dataset?
    There were 1267 rows and 24 columns.
3. What types of patient information are included in the dataset?
    It includes, sex, age, arrival mode, injury, chief complaint,mental state, pain, NRS_pain, vital signs,KTAS RN,Diagnosis in ED,            Disposition,KTAS expert,Error_group,Length of stay in mins, KTAS duration in mins,mistriage.
4. What are the mean age and maximum age?
    The mean was 54.42 and the max was 96.
5. How many patients are aged 65 or older?
    There are 467 senior patients.
6. How many patients have an oxygen saturation below 95?
    There are 47 patients with a SpO2 below 95, and one of them had a SpO2 of 20, so thats very bad. 
7. Which expert KTAS level appears most frequently?
    KTAS level 3.
8. What was the most difficult part of examining the dataset?
    The fact that some of the numbers were actually strings and probaly trying to understand how a notebook actually works.
    Also the file it's self was a little wonky, and tbh, idek how my advisor fixed it but she did so thats very cool. 